core/storage/base_model/gae_models.py

```diff
@@ -613,17 +613,22 @@ def _reconstitute(self, snapshot_dict):
         self.populate(**snapshot_dict)
         return self
 
-    def _reconstitute_from_snapshot_id(self, snapshot_id):
+    def _reconstitute_from_snapshot_id(self, snapshot_id, strict=True):
```

Comment:
This seems like a very general change. Why are we changing it in this PR, and is it actually necessary to do so?

Reply:
Then I would say -- it's OK to have strict=True in get_version(), but do not do it here. Handle the exception in get_version() as needed. In general one would not expect a method named like this to error out.

Reply:
Done.